### PR TITLE
microfetch: init at 0.3.3

### PR DESCRIPTION
**Pull request by nydragon**
## Description of changes

Addition of [microfetch]:

> A stupidly simple fetch tool, written in Rust. Runs in a fraction of a
> second, displays most nonsense people on r/unixporn care about. Aims
> to replace fastfetch on my system, but probably not on yours. Though
> you are more than welcome to use it on your system: it's fast.

[microfetch]: https://github.com/NotAShelf/microfetch

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]
LGTM

I'd propose adding an updateScript, but I am yet to release any tagged releases. We may revisit the derivation after I set up a CI for tagged builds.

File: pkgs/by-name/mi/microfetch/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "NotAShelf";
+    repo = "microfetch";
+    rev = "72d3130aa098a9e99dbca264a96e2372a1740204";
+    hash = "sha256-4l5EWjvueSF4ujNvW867HwVO/YoW3e1WLE7f+qlJ0aE=";
+  };
+
+  cargoHash = "sha256-d0S51K8f7w3tjDJMXinKKpxQKs3jgFFC0fhP/VDrU8I=";
+
+  meta = {
+    description = "A microscopic fetch script in Rust, for NixOS systems";
```
Comment (NotAShelf) [inline pkgs/by-name/mi/microfetch/package.nix:20]: Packaging guidelines dictate that description cannot start with indefinite articles.
```suggestion
    description = "Microscopic fetch script in Rust, for NixOS systems";
```

Comment (NotAShelf) [reply pkgs/by-name/mi/microfetch/package.nix:20]: I've also updated the repo description, since it doesn't make too much of a difference.

Comment (Nydragon) [reply pkgs/by-name/mi/microfetch/package.nix]: Just pushed the change.

On a side note: Is there a way to accept suggestions without creating a separate commit, i.e. amend the most recent commit?

Comment (NotAShelf) [reply pkgs/by-name/mi/microfetch/package.nix]: Not to my knowledge.

**Comment by NotAShelf** [comment]
I've officially released a tag (v0.3.0) which should be the last one for a while, until I can squeeze in more performance somehow - we may use the tagged release and add an update script since future releases will be tagged as well.

**Comment by Nydragon** [comment]
> I've officially released a tag (v0.3.0) which should be the last one for a while, until I can squeeze in more performance somehow - we may use the tagged release and add an update script since future releases will be tagged as well.

I have updated the tracked ref

File: pkgs/by-name/mi/microfetch/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "microfetch";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "NotAShelf";
+    repo = "microfetch";
+    rev = "v${version}";
```
Comment (NotAShelf) [inline pkgs/by-name/mi/microfetch/package.nix:13]:
```suggestion
    rev = "refs/tags/v${version}";
```

I'm not a huge fan of `rec` but this'll do.

Comment (Nydragon) [reply pkgs/by-name/mi/microfetch/package.nix]: Applied, I can replace `rec` by `let in` if that's what you had in mind.

Comment (NotAShelf) [reply pkgs/by-name/mi/microfetch/package.nix]: It was, but thats a whole debate on its own. I'm not going to push in either direction, so use what you prefer.

File: pkgs/by-name/mi/microfetch/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "microfetch";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "NotAShelf";
+    repo = "microfetch";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7pCy8x0HiQeuNZ3/0w4CyBkcrwn+xSuZBmrU3fZIC/c=";
```
Comment (donovanglover) [inline pkgs/by-name/mi/microfetch/package.nix:14]:
```suggestion
    hash = "sha256-iBfnui7xrx31euYnQxoJf1xXFYFiJnDMadLRAiRCN04=";
```

File: pkgs/by-name/mi/microfetch/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  cargoHash = "sha256-VYLKtuC0lv7TcZbGcWTdEYbtLtAnuYGU2RZ0pbaZtKM=";
```
Comment (donovanglover) [inline pkgs/by-name/mi/microfetch/package.nix:17]:
```suggestion
  cargoHash = "sha256-J+S6XIMUcWj4jXABQinEDx6dRG5Byc2UxJoKj2y1tQU=";
```

File: pkgs/by-name/mi/microfetch/package.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
```
Comment (donovanglover) [inline pkgs/by-name/mi/microfetch/package.nix:5]:
```suggestion
}:

```
Not an issue but we usually put a new line here

File: pkgs/by-name/mi/microfetch/package.nix (hunk 3 above)
Comment (Nydragon) [reply pkgs/by-name/mi/microfetch/package.nix]: I addressed all issues, thank you for reviewing.

File: pkgs/by-name/mi/microfetch/package.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  cargoHash = "sha256-J+S6XIMUcWj4jXABQinEDx6dRG5Byc2UxJoKj2y1tQU=";
+
+  meta = {
+    description = "Microscopic fetch script in Rust, for NixOS systems";
+    homepage = "https://github.com/NotAShelf/microfetch";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ nydragon ];
+    mainProgram = "microfetch";
```
Comment (donovanglover) [inline pkgs/by-name/mi/microfetch/package.nix:25]:
```suggestion
    mainProgram = "microfetch";
    platforms = lib.platforms.linux;
```
https://github.com/NotAShelf/microfetch/blob/927f6077b4b78cd112bc6932aab5287edc413ce8/flake.nix#L9

Comment (Nydragon) [reply pkgs/by-name/mi/microfetch/package.nix:25]: Fixed :+1:

**Review by donovanglover — APPROVED** [review]
LGTM

**Pull request merged**